### PR TITLE
Add vector radial search blog

### PR DESCRIPTION
**Pull request by kolchfa-aws**
Add vector radial search blog

### Check List
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the BSD-3-Clause License.

### CODE REVIEW

File: _posts/2024-06-20-vector-radial-search.md [hunk 1]
```diff
@@ -0,0 +1,360 @@
...
+
+Before OpenSearch version 2.14, the OpenSearch k-NN plugin only provided top-k queries for approximate vector similarity search. OpenSearch 2.14 introduced a new type of vector search---_radial search_. Radial search is now supported in both [k-NN search]((https://opensearch.org/docs/latest/search-plugins/knn/radial-search-knn/)) and [neural search](https://opensearch.org/docs/latest/query-dsl/specialized/neural/) also support radial search.
+
+Radial search enhances the capabilities of the OpenSearch k-NN plugin beyond approximate top-k searches. With radial search, you can search for all points in a vector space that are within a specified maximum distance or minimum score threshold from a query point. This provides increased flexibility and utility in search operations. 
+
+For instance, if you want to search for documents most similar to a given document, you can run a query to retrieve the top K most similar documents (where K = top N). However, if K is not chosen optimally, you might receive irrelevant results or miss relevant ones. With vector radial search, you can run a query to retrieve all highly similar documents by using a threshold (e.g., similarity score > 0.95).
```
Comment (natebower) [inline _posts/2024-06-20-vector-radial-search.md:22]: Above we use "top-k", but here we use "top K" and "K".

Comment (natebower) [inline _posts/2024-06-20-vector-radial-search.md:22]:
```suggestion
For instance, if you want to search for documents most similar to a given document, you can run a query to retrieve the top K most similar documents (where K = top N). However, if K is not chosen optimally, you might receive irrelevant results or miss relevant ones. With vector radial search, you can run a query to retrieve all highly similar documents by using a threshold (for example, similarity score > 0.95).
```

File: _posts/2024-06-20-vector-radial-search.md [hunk 2]
```diff
@@ -0,0 +1,360 @@
...
+
+    The following image shows an example of radial search with the [L2 space type](https://opensearch.org/docs/latest/search-plugins/knn/radial-search-knn/#spaces) and `max_distance` = 25.0. The number within each circle represents the distance of the point from the query target. All points within the `max_distance` of 25.0 are included in the results.
+
+    <img src="/assets/media/blog-images/2024-06-20-vector-radial-search/radial-search-with-max-distance.png" alt="Radial search with max distance" class="center"/>{:style="width: 100%; max-width: 800px; height: auto; text-align: center"}
+
+- **Minimum score**: Define a similarity score threshold, retrieving all points that meet or exceed that score in relation to your query point. This is useful when relative similarity, based on a specific metric, is more critical than physical proximity. 
+   
+    The following image shows a example of radial search with `min_score`= 0.90. The number within each circle represents the point's OpenSearch score relative to the query target. All points whose score is 0.90 or above are included in the results.
```
Comment (natebower) [inline _posts/2024-06-20-vector-radial-search.md:36]:
```suggestion
    The following image shows an example of radial search with `min_score`= 0.90. The number within each circle represents the point's OpenSearch score relative to the query target. All points whose score is 0.90 or above are included in the results.
```

File: _posts/2024-06-20-vector-radial-search.md [hunk 3]
```diff
@@ -0,0 +1,360 @@
...
+
+    <img src="/assets/media/blog-images/2024-06-20-vector-radial-search/radial-search-with-min-score.png" alt="Radial search with min score" class="center"/>{:style="width: 100%; max-width: 800px; height: auto; text-align: center"}
+
+## When to use radial search
+
+Use radial search instead of top-k vector search in the following scenarios:
+
+* **Proximity-based filtering**: When you need to find all items within a specific distance from a query point. Unlike traditional k-NN searches that return a fixed number of top results,
+radial search allows you to find all items that fall within a specified distance threshold.
+* **Threshold-specific queries**: When you need to make sure only items that meet your criteria are included in the results. Radial search lets you define specific similarity or distance thresholds, which is essential for tasks like anomaly detection and geospatial searches.
```
Comment (natebower) [inline _posts/2024-06-20-vector-radial-search.md:46]:
```suggestion
* **Threshold-specific queries**: When you need to make sure that only items that meet your criteria are included in the results. Radial search lets you define specific similarity or distance thresholds, which is essential for tasks like anomaly detection and geospatial searches.
```

File: _posts/2024-06-20-vector-radial-search.md [hunk 4]
```diff
@@ -0,0 +1,360 @@
...
+* **Dynamic range adjustments:** When the acceptable similarity or distance range can vary. Radial search offers more flexible search results.
+
+## Supported configurations
+
+Starting with OpenSearch 2.14, you can perform radial search with either the Lucene or Faiss engines in OpenSearch k-NN plugin. The following table summarizes radial search use cases by engine.
```
Comment (natebower) [inline _posts/2024-06-20-vector-radial-search.md:51]:
```suggestion
Starting in OpenSearch 2.14, you can perform radial search with either the Lucene or Faiss engines using the OpenSearch k-NN plugin. The following table summarizes radial search use cases by engine.
```

File: _posts/2024-06-20-vector-radial-search.md [hunk 5]
```diff
@@ -0,0 +1,360 @@
...
+{"my_vector": [7.1, 7.4], "price": 14.2}
+{"index": {"_index": "knn-index-test", "_id": "3"}}
+{"my_vector": [7.3, 8.3], "price": 19.1}
+{"index": {"_index": "knn-index-test", "_id": "4"}}
+{"my_vector": [6.5, 8.8], "price": 1.2}
+{"index": {"_index": "knn-index-test", "_id": "5"}}
+{"my_vector": [5.7, 7.9], "price": 16.5}
+```
+
+### Radial search with max distance and a filter
```
Comment (natebower) [inline _posts/2024-06-20-vector-radial-search.md:116]:
```suggestion
### Radial search with a max distance and a filter
```

File: _posts/2024-06-20-vector-radial-search.md [hunk 6]
```diff
@@ -0,0 +1,360 @@
...
+                    "my_vector": [6.5,8.8],
+                    "price": 1.2
+                }
+            }
+        ]
+    }
+}
+```
+
+### Radial search with min score and a filter
```
Comment (natebower) [inline _posts/2024-06-20-vector-radial-search.md:177]:
```suggestion
### Radial search with a min score and a filter
```

File: _posts/2024-06-20-vector-radial-search.md [hunk 7]
```diff
@@ -0,0 +1,360 @@
...
+            ]
+          }
+        }
+      }
+    }
+  }
+}
+```
+
+The following example shows a k-NN radial search with `min_score` of `0.95` and the same filter as the preceding query:
```
Comment (natebower) [inline _posts/2024-06-20-vector-radial-search.md:269]:
```suggestion
The following example shows a k-NN radial search with a `min_score` of `0.95` and the same filter as the preceding query:
```

File: _posts/2024-06-20-vector-radial-search.md [hunk 8]
```diff
@@ -0,0 +1,360 @@
...
+    }
+  }
+}
+```
+
+## Benchmarks
+
+We ran radial search benchmarks using the following cluster configuration.
+
+|OpenSearch version	|Leader nodes	|Leader node type	|Leader node disk space	|Data nodes	|Data node type	|Data node disk space	|Primary shard count	|Replica count	|Availability zone	|Test client type	|
```
Comment (natebower) [inline _posts/2024-06-20-vector-radial-search.md:309]: If "Availability zone" refers to an AWS Availability Zone, then "Zone" should be capitalized.

File: _posts/2024-06-20-vector-radial-search.md [hunk 9]
```diff
@@ -0,0 +1,360 @@
...
+
+For benchmarking, we used [OpenSearch Benchmark](https://opensearch.org/docs/latest/benchmark/).
+
+### Dataset
+
+We used the following dataset for benchmarking.
+
+|Name	|Dimension	|Doc size	|Query size	|Space type	|
+|---	|---	|---	|---	|---	|
+|cohere-wikipedia-22-12-en-embeddings	|768	|1M	|10k	|inner product	|
```
Comment (natebower) [inline _posts/2024-06-20-vector-radial-search.md:323]: Should "Inner" be capitalized?

File: _posts/2024-06-20-vector-radial-search.md [hunk 10]
```diff
@@ -0,0 +1,360 @@
...
+
+The dataset was updated to include the top-k threshold values and true neighbors for radial threshold. In our benchmarks, top-k refers to the number of nearest neighbors considered when setting the minimum score and maximum distance thresholds for radial search. For instance, a top-k 100 setting means we used the 100th closest document to a query point to determine these thresholds. The following table summarizes the threshold configuration.
```
Comment (natebower) [inline _posts/2024-06-20-vector-radial-search.md:325]:
```suggestion
The dataset was updated to include the top-k threshold values and true neighbors for the radial threshold. In our benchmarks, top-k refers to the number of nearest neighbors considered when setting the minimum score and maximum distance thresholds for radial search. For instance, a top-k 100 setting means that we used the 100th closest document to a query point to determine these thresholds. The following table summarizes the threshold configuration.
```

Comment (natebower) [inline _posts/2024-06-20-vector-radial-search.md:325]: Should the first instance of "top-k" be either in quotation marks or italics?

File: _posts/2024-06-20-vector-radial-search.md [hunk 11]
```diff
@@ -0,0 +1,360 @@
...
+
+|Threshold name	|Min score threshold	|Median num of true neighbors	|Average num of true neighbors	|Num of query targets having 0 neighbors	|
```
Comment (natebower) [inline _posts/2024-06-20-vector-radial-search.md:327]:
```suggestion
|Threshold name	|Min score threshold	|Median num of true neighbors	|Average num of true neighbors	|Num of query targets with 0 neighbors	|
```

File: _posts/2024-06-20-vector-radial-search.md [hunk 12]
```diff
@@ -0,0 +1,360 @@
...
+|---	|---	|---	|---	|---	|
+|threshold1	|161	|118	|83.1888	|4048	|
+|threshold2	|156	|1186	|421.8486	|1661	|
+|threshold3	|154	|2959	|778.0673	|941	|
+
+### Algorithm
+
+The algorithm used was specified as follows.
```
Comment (natebower) [inline _posts/2024-06-20-vector-radial-search.md:335]: Is "specified" the right word here, or do we mean "configured"?

File: _posts/2024-06-20-vector-radial-search.md [hunk 13]
```diff
@@ -0,0 +1,360 @@
...
+|min score threshold2	|Faiss	|min_score	|7.76	|13.58	|26.83	|0.98	|
+|min score threshold3	|Faiss	|min_score	|6.89	|12.13	|25.59	|0.98	|
+|min score threshold1	|Lucene	|min_score	|4.42	|17.01	|50.14	|0.85	|
+|min score threshold2	|Lucene	|min_score	|13.02	|54.16	|118.65	|0.90	|
+|min score threshold3	|Lucene	|min_score	|22.56	|84.5	|161.62	|0.92	|
+
+
+## Summary
+
+OpenSearch 2.14 introduced vector radial search, significantly expanding the capabilities of the k-NN plugin. This powerful feature allows searches based on distance or score thresholds, offering greater flexibility for various applications. Supported by both Lucene and Faiss engines, radial search caters to diverse use cases.
```
Comment (natebower) [inline _posts/2024-06-20-vector-radial-search.md:358]:
```suggestion
OpenSearch 2.14 introduced vector radial search, significantly expanding the capabilities of the k-NN plugin. This powerful feature allows searches based on distance or score thresholds, offering greater flexibility for various applications. Supported by both the Lucene and Faiss engines, radial search caters to diverse use cases. 
```

**Review by natebower — COMMENTED** [review]
@kolchfa-aws Thanks so much! Please see my comments and changes and let me know if you have any questions.

@pajuric This will be ready to publish once @kolchfa-aws indicates that my comments and changes have been addressed.

File: _posts/2024-06-20-vector-radial-search.md (hunk 8 above)
Comment (kolchfa-aws) [inline _posts/2024-06-20-vector-radial-search.md:309]:
```suggestion
|OpenSearch version	|Leader nodes	|Leader node type	|Leader node disk space	|Data nodes	|Data node type	|Data node disk space	|Primary shard count	|Replica count	|Availability Zone	|Test client type	|
```

File: _posts/2024-06-20-vector-radial-search.md (hunk 9 above)
Comment (kolchfa-aws) [inline _posts/2024-06-20-vector-radial-search.md:323]:
```suggestion
|cohere-wikipedia-22-12-en-embeddings	|768	|1M	|10k	|Inner product	|
```

File: _posts/2024-06-20-vector-radial-search.md (hunk 12 above)
Comment (kolchfa-aws) [inline _posts/2024-06-20-vector-radial-search.md:335]:
```suggestion
The algorithm used was configured as follows.
```

File: _posts/2024-06-20-vector-radial-search.md (hunk 1 above)
Comment (kolchfa-aws) [reply _posts/2024-06-20-vector-radial-search.md]: Changed to "top K"

File: _posts/2024-06-20-vector-radial-search.md [hunk 14]
```diff
@@ -0,0 +1,360 @@
...
+|HNSW	|256	|256	|16	|
+
+### Results
+
+The following table presents the benchmarking results.
+
+|Query threshold	|Engine type	|Query type	|Search: 50th percentile service time (ms)	|Search: 90th percentile service time (ms)	|Search: 99th percentile service time (ms)	|Recall	|
+|---	|---	|---	|---	|---	|---	|---	|
+|min score threshold1	|Faiss	|min_score	|6.24	|7.33	|11.11	|0.99	|
+|---	|---	|---	|---	|---	|---	|---	|
```
Comment (junqiu-lei) [inline _posts/2024-06-20-vector-radial-search.md:348]: Do we need this line?

**Comment by junqiu-lei** [comment]
Overall looks good to me, thanks so much for the PR!

File: _posts/2024-06-20-vector-radial-search.md [hunk 15]
```diff
@@ -0,0 +1,360 @@
...
+* **Dynamic range adjustments:** When the acceptable similarity or distance range can vary. Radial search offers more flexible search results.
+
+## Supported configurations
+
+Starting in OpenSearch 2.14, you can perform radial search with either the Lucene or Faiss engines using the OpenSearch k-NN plugin. The following table summarizes radial search use cases by engine.
+
+|Engine	|Filter supported	|Nested field supported	|Search type	|
+|---	|---	|---	|---	|
+|Lucene	|Yes	|No	|Approximate	|
+|---	|---	|---	|---	|
```
Comment (kolchfa-aws) [inline _posts/2024-06-20-vector-radial-search.md:56]:
```suggestion
```

File: _posts/2024-06-20-vector-radial-search.md (hunk 14 above)
Comment (kolchfa-aws) [inline _posts/2024-06-20-vector-radial-search.md:348]:
```suggestion
```

Comment (kolchfa-aws) [reply _posts/2024-06-20-vector-radial-search.md]: No :) Removed.

File: _posts/2024-06-20-vector-radial-search.md [hunk 16]
```diff
@@ -0,0 +1,358 @@
...
+title:  Understanding vector radial search in OpenSearch
+authors:
+  - junqiu
+  - vamshin
+  - dylantong
+date: 2024-06-20
+categories:
+    - technical-posts
+meta_keywords: 
+meta_description:
```
Comment (pajuric) [inline _posts/2024-06-20-vector-radial-search.md:12]: Please update the PR with the following meta: 

_meta_keywords:_ vector radial search, OpenSearch k-NN plugin, k-NN search, neural search

_meta_description:_  Discover how vector radial search enhances the capabilities of the OpenSearch k-NN plugin and increases flexibility and utility in search operations.

File: _posts/2024-06-20-vector-radial-search.md [hunk 17]
```diff
@@ -0,0 +1,358 @@
+---
+layout: post
+title:  Understanding vector radial search in OpenSearch
+authors:
+  - junqiu
+  - vamshin
+  - dylantong
+date: 2024-06-20
```
Comment (pajuric) [inline _posts/2024-06-20-vector-radial-search.md:8]: Please update the publish date to 2024-06-24

Comment (pajuric) [reply _posts/2024-06-20-vector-radial-search.md:8]: @vamshin Please update the blog meta and date. I'd like to distribute this today.

**Comment by pajuric** [comment]
@krisfreedain - This blog can push later today one the meta is updated.

File: _posts/2024-06-20-vector-radial-search.md (hunk 17 above)
Comment (junqiu-lei) [reply _posts/2024-06-20-vector-radial-search.md:8]: Maybe @kolchfa-aws could help update the blog meta and date? Thanks!

Comment (krisfreedain) [reply _posts/2024-06-20-vector-radial-search.md]: I took care of the edits.

**Review by krisfreedain — APPROVED** [review]
looks good!

**Pull request merged**